### DISCUSSION TITLE
Storing information for each element in mesh file

### DISCUSSION THREAD

**Discussion by Sina-av** [idaholab/moose#26712] [Q&A Modules: Solid mechanics]
Hi there,
I want to run a mechanical problem where each element of my mesh has a different Young's modulus.
I.e., I want to have a mesh that can store the value of Young's modulus for each single element. Later I want to access all those values in my material file in Moose.
I was wondering if there is a mesh format (or a way in general) to do so.
I know that while using .e file there is the option for "extra element integer", but this option only stores an integer which might not be helpful for storing material parameters in cells.
Thank you so much.

**Comment by GiudGiud** [comment]
Hello

How is this young modulus computed? Do you just have a massive CSV file?
Or do you have an analytical expression?

there's plenty of options here for having an element-wise Young modulus

Guillaume

**Reply by Sina-av** [reply]
Hi Guillaume,
At this stage I want to have random Young's modulus for each element. I was thinking about using tools like GSTOOLs (https://github.com/GeoStat-Framework/GSTools), where an input mesh can be given and then a random field is added to that mesh.
But I am also looking into other methods too. So please let me know if you have any ideas in general.
Thank so much again.

**Reply by GiudGiud** [reply]
for a first try you can use a CSV file
then a propertyReadFile to read that file (there s a certain format in the csv)
then a PiecewiseFunctionFromCSV to get it into a field

**Reply by Sina-av** [reply]
thanks so much. I will try this and will keep you updated.
Best.
